web/config/wikijump.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+return [
+
+    /**
+     * The required length for a username.
+     */
+
+    'username_min' => 1,
+    'username_max' => 20,
+
+    /**
+     * How many username changes a user is allowed.
+     * A farm admin can force changes above this limit.
+     */
+    'username_change_limit' => 3,
+
+    /**
+     * Forbidden usernames.
+     */
+    'forbidden_usernames' => [
```

Comment:
Should we think about adding a basic curse word filter here too? I'm sure there is a decent library for it - I'd rather make it harder than easier for people to use the n-word for their name.

Reply:
There is a separate issue for that: [WJ-260](https://scuttle.atlassian.net/browse/WJ-260).

To allow sites to add their own restriction lists (for instance, for non-English languages) it should be a feature that hits the database. We can add some general blacklists here, but the comprehensive filtering is separate.